### ISSUE TITLE
NullPointerException: softwareVersion must not be null

### ISSUE THREAD

**Issue by AliSoftware** [wordpress-mobile/WordPress-Android#15278]
**Sentry Url**: https://sentry.io/share/issue/1f3cd26edc6f4e85826fd55050588d8c/
**User Count**: 69
**Count**: 400
**First Release**: 18.0-rc-1
**First Seen**: 2021-08-14T01:52:02.111000Z
**Last Seen**: 2021-09-03T11:17:34Z
**24 Hours**: 23
**30 Days**: 400

```
NullPointerException: softwareVersion must not be null
     at org.wordpress.android.fluxc.store.EditorThemeStore.hasRequiredWordPressVersion(EditorThemeStore.kt:196)
     at org.wordpress.android.fluxc.store.EditorThemeStore.editorSettingsAvailable(EditorThemeStore.kt:184)
     at org.wordpress.android.fluxc.store.EditorThemeStore.access$editorSettingsAvailable(EditorThemeStore.kt:32)
     at org.wordpress.android.fluxc.store.EditorThemeStore$onAction$1.invokeSuspend(EditorThemeStore.kt:77)
     at org.wordpress.android.fluxc.store.EditorThemeStore$onAction$1.invoke(null:10)
     at org.wordpress.android.fluxc.tools.CoroutineEngine$launch$1.invokeSuspend(CoroutineEngine.kt:43)
...
(3 additional frame(s) were not displayed)
```

**Comment by develric** [comment]
Was not able to reproduce on my side unless forcing the null value by debugger of course; the issue seems to be that [softwareVersion](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/a471f6245ea15181c3463a1892a3a586ac546c83/fluxc/src/main/java/org/wordpress/android/fluxc/store/EditorThemeStore.kt#L196) in kotlin is seen  and managed as not nullable since the return value in [getSoftwareVersion](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/a471f6245ea15181c3463a1892a3a586ac546c83/fluxc/src/main/java/org/wordpress/android/fluxc/model/SiteModel.java#L300) in SiteModel is not marked as @Nullable but can be null eventually.

Hi @antonis 👋 , if I'm not wrong this could be related to [this](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2049) FluxC PR. Would you mind having a look?

Numbers are currently low (in 18.0 is affecting like 0.01% of users if I'm not wrong with the maths) and given it's there from 18.0 I would expect numbers to stay steady, but given it's a crash I'm assigning a medium priority. I will monitor the numbers during the week in case they could rise. 

In case we want to make a fix through current 18.2 beta, linking here this @AliSoftware slack note for the branch to target (p1630919748011100?thread_ts=1630919206.008900&cid=C012H19SZQ8-slack-wp-crash-triage)

**Comment by antonis** [comment]
Resolved in `18.2-rc-2` with https://github.com/wordpress-mobile/WordPress-Android/pull/15286
